compiler/ras/DebugCounter.md

```diff
@@ -0,0 +1,126 @@
+
+For example:
+
+```
+prependDebugCounter("frames", tt);
+...
+prependDebugCounter("frames:#autos", tt, numAutos);
+```
+
+This produces two counters that are unrelated except that the jit will report their ratio when the run is complete:
+
+```
+ 1: frames         |  3914036134 |   __   1 __
+ 2: frames:#autos  | 22680880758 |      5.79  |  __   2 __
+```
+
+### Equals `=`
+
+The equals sign indicates to the debug counter display routine that the remainder of the counter name should be **sorted numerically** rather than alphabetically. It has no other special meaning.
+
+### Parentheses `()`
+
+Parentheses are used to mark a **verbatim section** of a counter name, within which the other special characters lose their special meaning. For example, asking for a counter with a name of the form "(a/b)" produces just one counter. Without the special meaning of parentheses, this would have produced two counters: "(a" and "(a/b)".
+
+Parentheses can be nested.
+
+### Hash `#`
+
+The hash symbol has no particular special meaning, but is used by convention to signify counters that are **suitable for producing a histogram**. Generally you can include a hash symbol in any debug counter that is incremented by some value other than 1, such as "#parameters".
+
+### Other naming considerations
+
+Counters can have arbitrary names, though care should be taken with certain special characters:
+
+* Whitespace and other shell-control characters may make them awkward to specify on the command line
+* Characters with special meanings in Testarossa's [regular expressions](../../doc/compiler/regular_expression/Regular_Expression.md) will make them awkward to filter
+
+## Specifying Output
+
+### Environment Variables
+
+#### `TR_DebugCounterFileName`
+
+Specifies the file to which the debug counter output should be appended (creating the file if necessary), relative to the current working directory. If not specified, debug counter output is written to stdout.
+
+### JIT Options
+
+At least one debug counter option must be specified for debug counter output to be produced.
+
+#### `debugCounters=`
+
+Specifies a [regular expression](../../doc/compiler/regular_expression/Regular_Expression.md) that is matched against dynamic debug counter names to determine which are included in the output.
+
+#### `staticDebugCounters=`
+
+Specifies a [regular expression](../../doc/compiler/regular_expression/Regular_Expression.md) that is matched against static debug counter names to determine which are included in the output.
+
+### Example
+
+```
+export TR_DebugCounterFileName=debugcounters.txt
+```
+Setting the option to `debugCounters={*}` will output all dynamic debug counters.
```

Comment:
This is not a useful thing in general due to the nature of debug counters and is probably not worth documenting.